### PR TITLE
Let's you place drinks, glasses, and shakers inside chem/booze/soda/etc dispensers

### PR DESCRIPTION
**Pull request by JMWTurner**
Not being able to put my shaker in the booze dispenser triggered me earlier today.

### CODE REVIEW

**Comment by Exxion** [comment]
Closes #9192

**Comment by ComicIronic** [comment]
You need to fix the type of the beaker var, as well as everywhere it is assumed to be a beaker. You should also probably rename it.

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -270,7 +270,7 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 		if(!can_use(user))
 			return
 
-	if(istype(D, /obj/item/weapon/reagent_containers/glass))
+	if(istype(D, /obj/item/weapon/reagent_containers/glass) || istype(D, /obj/item/weapon/reagent_containers/food/drinks))
 		if(src.beaker)
 			to_chat(user, "A beaker is already loaded into the machine.")
```
Comment (ComicIronic) [inline code/modules/reagents/machinery/chem_dispenser.dm]: You should replace beaker here as well. Ideally with the loaded name, a la `\A [src.beaker]`

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -12,7 +12,7 @@
 	var/max_energy = 50
 	var/rechargerate = 2
 	var/amount = 30
-	var/obj/item/weapon/reagent_containers/glass/beaker = null
+	var/obj/item/weapon/reagent_containers/glass/container = null
```
Comment (ComicIronic) [inline code/modules/reagents/machinery/chem_dispenser.dm]: This type is wrong, it should just be a reagent_containers.

**Pull request merged**